src/Model/Storage.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Kirby\Model;
+
+use Kirby\Data\Data;
+use Kirby\Exception\LogicException;
+use Throwable;
+
+abstract class Storage
+{
+	public function __construct(
+		protected Model $model,
+	) {}
+
+	abstract protected static function contentFile($meta): string;
+
+	public function children(): array
+	{
+		throw new LogicException('This model does not have children');
+	}
+
+	public function files(): array
+	{
+		throw new LogicException('This model does not have files');
+	}
+
+	abstract public static function find(string $class, string $identifier): Model|null;
+
+	public function read(): array
+	{
+		try {
+			return Data::read($this->contentFile($this->model->meta()));
+		} catch (Throwable $e) {
+			return [];
+		}
+	}
+
+	public function changeMeta(array $meta): ModelMeta
```

Comment:
`Suggestion`: We could already pass `ModelMeta $meta` here so that the storage handler has a cleaner interface. It shouldn't be the handler's obligation to construct the meta object. Also it would allow the handler to return `void` when changing meta (consistent with the existing methods for updating properties).